datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -83,6 +86,53 @@ impl ScalarFunctionExpr {
         }
     }
 
+    /// Create a new Scalar function
+    pub fn try_new(
+        fun: Arc<ScalarUDF>,
+        args: Vec<Arc<dyn PhysicalExpr>>,
+        schema: &Schema,
+    ) -> Result<Self> {
+        let name = fun.name().to_string();
+        let arg_types = args
+            .iter()
+            .map(|e| e.data_type(schema))
+            .collect::<Result<Vec<_>>>()?;
+
+        // verify that input data types is consistent with function's `TypeSignature`
+        data_types_with_scalar_udf(&arg_types, &fun)?;
+
+        let nullables = args
+            .iter()
+            .map(|e| e.nullable(schema))
+            .collect::<Result<Vec<_>>>()?;
+
+        let arguments: Vec<String> = args
+            .iter()
+            .map(|e| {
+                e.as_any()
+                    .downcast_ref::<Literal>()
+                    .map(|literal| match literal.value() {
+                        ScalarValue::Utf8(Some(s)) => s.clone(),
```

Comment:
It would be really nice to avoid this clone here -- if we changed the signature to `&[Option<&str>]` I think that would be easier to use and more efficient